rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/robolectric/fragment/CovidOtherClinicalDataFragmentTest.java

```diff
@@ -0,0 +1,86 @@
+package io.ona.rdt.robolectric.fragment;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.robolectric.Robolectric;
+import org.robolectric.util.ReflectionHelpers;
+
+import io.ona.rdt.R;
+import io.ona.rdt.activity.CovidPatientProfileActivity;
+import io.ona.rdt.domain.Patient;
+import io.ona.rdt.fragment.CovidOtherClinicalDataFragment;
+import io.ona.rdt.presenter.CovidOtherClinicalDataFragmentPresenter;
+import io.ona.rdt.robolectric.RobolectricTest;
+import io.ona.rdt.util.Constants;
+import io.ona.rdt.util.CovidConstants;
+
+public class CovidOtherClinicalDataFragmentTest extends RobolectricTest {
+
+    @Mock
+    private CovidOtherClinicalDataFragmentPresenter presenter;
+
+    private CovidOtherClinicalDataFragment covidOtherClinicalDataFragment;
+    private Patient patient;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        covidOtherClinicalDataFragment = buildFragment();
+        ReflectionHelpers.setField(covidOtherClinicalDataFragment, "presenter", presenter);
+    }
+
+    @Test
+    public void testOnClick() {
+        View view = Mockito.mock(View.class);
+
+        Mockito.when(view.getId()).thenReturn(R.id.tv_covid_wbc);
+        covidOtherClinicalDataFragment.onClick(view);
+
+        Mockito.verify(presenter).launchForm(covidOtherClinicalDataFragment.getActivity(), CovidConstants.Form.WBC_FORM, patient);
+    }
+
+    @Test
+    public void testOnDestroyView() {
+        View patientProfileTabbedFragmentContainer = covidOtherClinicalDataFragment.getActivity().findViewById(R.id.patient_profile_tabbed_fragment_container);
+        View patientProfileFragmentContainer = covidOtherClinicalDataFragment.getActivity().findViewById(R.id.patient_profile_fragment_container);
+
+        Assert.assertEquals(View.GONE, patientProfileTabbedFragmentContainer.getVisibility());
+        Assert.assertEquals(View.VISIBLE, patientProfileFragmentContainer.getVisibility());
+
+        covidOtherClinicalDataFragment.onDestroyView();
+
+        Assert.assertEquals(View.VISIBLE, patientProfileTabbedFragmentContainer.getVisibility());
+        Assert.assertEquals(View.GONE, patientProfileFragmentContainer.getVisibility());
+    }
+
+    private CovidOtherClinicalDataFragment buildFragment() {
```

Comment:
Could we use a [fragmentScenario](https://github.com/onaio/rdt-standard/blob/475d773834a237dcbcc2e993a0fdb4da4ad8f86c/rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/robolectric/fragment/CovidPatientVisitFragmentTest.java#L36) instead?

Reply:
`CovidOtherClinicalDataFragment.java` calling `getActivity()` method in fragment `onCreate` method so if I used FragmentScenario so the host activity will remain empty for that reason test will fail with `NullPointerException` that's why I have used this approach.
But we can change this if somehow we attach the host activity to the fragment.

Reply:
I think we can find a way to do this with fragmentscenario. Could you do some research on this?

Reply:
the official doc says `Your fragment shouldn't be dependent on a specific parent activity or fragment.` https://developer.android.com/guide/fragments/test

Reply:
Have you tried using a [fragmentfactory](https://developer.android.com/reference/androidx/fragment/app/FragmentFactory) to inject a dummy activity?

Reply:
fixed with fragment factory